torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -383,9 +384,68 @@ std::string dump_nccl_trace(
 control_plane::RegisterHandler dumpHandler{
     "dump_nccl_trace_pickle",
     [](const control_plane::Request& req, control_plane::Response& res) {
-      // TODO: c-p-i-o: params from the request need to go to dump_nccl_trace.
+      auto params = req.params();
```

Comment:
nit: might be worth moving the handler registrations to a new .cpp file? or to NCCLUtils.cpp perhaps, idk.  You can think about it for later